src/engine/strat_engine/engine.rs

```diff
@@ -122,6 +141,42 @@ impl Engine for StratEngine {
     fn pools(&self) -> Vec<&Pool> {
         self.pools.into_iter().map(|x| x as &Pool).collect()
     }
+
+    fn get_eventable(&mut self) -> EngineResult<Option<Box<Eventable>>> {
+        Ok(Some(Box::new(DM::new()?)))
+    }
+
+    fn evented(&mut self) -> EngineResult<()> {
+        let device_list = DM::new()?.list_devices()?;
+
+        for &(ref dm_name, _, new_event_nr) in &device_list {
+            if let Some(new_event_nr) = new_event_nr {
+                if let Some(prev_event_nr) = self.watched_devs.get_mut(dm_name) {
+                    if new_event_nr > *prev_event_nr {
+                        // Event! We know the dm name, but don't know which
+                        // pool
+                        for pool in &mut self.pools {
+                            if pool.get_eventables().contains(dm_name) {
+                                pool.event_on(dm_name)?;
+                                break;
+                            }
+                        }
+                        *prev_event_nr = new_event_nr;
+                    }
+                }
+            }
+        }
+
+        self.watched_devs = device_list
```

Comment:
I don't thing these devs are really watched. Maybe ```last_observed_event_nrs``` would be a better name.

Reply:
yes, ok

Reply:
On second thought, they are. This is a list of only the dm names we care about, and the last seen event numbers for each. I'll try to name it something better, and we can see if it's better.